doc/user/content/overview/delta-joins.md

```diff
@@ -0,0 +1,432 @@
+
+```sql
+...
+    AND c_custkey = o_custkey
+    AND l_orderkey = o_orderkey
+...
+```
+
+So we have a foreign key `l_orderkey` in `lineitem` and another foreign key `o_custkey` in `orders`. The trick will be to define "narrow" views of `lineitem` and `orders` that contain only the primary key and foreign key, and build indexes for each.
+
+```sql
+-- Create a "narrow" view containing foreign key `l_orderkey` and `lineitem`'s composite primary key (l_orderkey, l_linenumber).
+CREATE VIEW lineitem_fk_orderkey AS SELECT l_orderkey, l_linenumber FROM lineitem;
+CREATE INDEX lineitem_fk_orderkey_0 ON lineitem_fk_orderkey (l_orderkey, l_linenumber);
+CREATE INDEX lineitem_fk_orderkey_1 ON lineitem_fk_orderkey (l_orderkey);
+-- Create a "narrow" view containing foreign key `o_custkey` and `orders`'s primary key `o_orderkey`.
+CREATE VIEW orders_fk_custkey AS SELECT o_orderkey, o_custkey FROM orders;
+CREATE INDEX orders_fk_custkey_0 on orders_key_custkey (o_orderkey);
+CREATE INDEX orders_fk_custkey_1 on orders_key_custkey (o_custkey);
+```
+
+With these new "narrow" views and their indexes, we can rewrite `query_03` to perform the core equijoin logic. We then join their primary keys back to the `orders` and `lineitem` collections, which are indexed only by their primary keys.
+
+```sql
+CREATE VIEW query_03_optimized AS
+SELECT
+    o.o_orderkey,
+    o.o_orderdate,
+    o.o_shippriority,
+    sum(l.l_extendedprice * (1 - l.l_discount)) AS revenue
+FROM
+    customer c,
+    orders o,
+    lineitem l,
+    -- NEW: "narrow" collections containing just primary and foreign keys.
+    lineitem_fk_orderkey l_ok,
+    orders_fk_custkey o_ck
+WHERE
+    -- core equijoin constraints using "narrow" collections.
+    o_ck.o_custkey = c.c_custkey
+    AND l_ok.l_orderkey = o.o_orderkey
+    -- connect narrow and wide collections.
+    AND o_ck.o_orderkey = o.o_orderkey
+    AND l_ok.l_orderkey = l.l_orderkey
+    AND l_ok.l_linenumber = l.l_linenumber
+    -- constraints on values from wide collections.
+    AND c.c_mktsegment = 'BUILDING'
+    AND o.o_orderdate < DATE '1995-03-15'
+    AND l.l_shipdate > DATE '1995-03-15'
+GROUP BY
+    o.o_orderkey,
+    o.o_orderdate,
+    o.o_shippriority;
+```
+
+Trigger computation by creating an index on the `query_03_optimized` view.
+
+```sql
+CREATE DEFAULT INDEX query_03_optimized_idx ON query_03_optimized;
+```
+
+What happens now in join planning is that "delta query" planning still kicks in. We have all of the necessary arrangements at hand to avoid maintaining intermediate state. The difference is that we only ever use one arrangement for each of the "wide" relations. The relations that must be arranged multiple times are narrow relations whose rows can be substantially smaller. You can confirm you are using a delta join by running `EXPLAIN VIEW query_03_optimized;` and noting that the output contains `type=delta`.
```

Comment:
```suggestion
What happens now in join planning is that "delta query" planning still kicks in. We have all the necessary indexes to avoid maintaining intermediate state. The difference is that we only ever use one index for each "wide" relation. The relations Materialize must index multiple times are narrow relations whose rows can be substantially smaller. You can confirm you are using a delta join by running `EXPLAIN VIEW query_03_optimized;` and noting that the output contains `type=delta`.
```